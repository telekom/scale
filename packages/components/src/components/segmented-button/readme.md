# scale-segmented-button-group



<!-- Auto Generated Below -->


## Properties

| Property               | Attribute                | Description                                                  | Type                             | Default                                  |
| ---------------------- | ------------------------ | ------------------------------------------------------------ | -------------------------------- | ---------------------------------------- |
| `ariaLabelTranslation` | `aria-label-translation` | (optional) aria-label attribute needed for icon-only buttons | `string`                         | ``segment button with $slottedSegments`` |
| `disabled`             | `disabled`               | (optional) If `true`, the button is disabled                 | `boolean`                        | `false`                                  |
| `fullWidth`            | `full-width`             | (optional) If `true`, expand to container width              | `boolean`                        | `false`                                  |
| `helperText`           | `helper-text`            | (optional) If `true`, show error message                     | `string`                         | `'Please select an option'`              |
| `iconOnly`             | `icon-only`              | (optional) icon only                                         | `boolean`                        | `false`                                  |
| `invalid`              | `invalid`                | (optional) If `true`, show error message                     | `boolean`                        | `false`                                  |
| `label`                | `label`                  | (optional) Button label                                      | `string`                         | `undefined`                              |
| `longestButtonWidth`   | `longest-button-width`   |                                                              | `string`                         | `undefined`                              |
| `multiSelect`          | `multi-select`           | (optional) Allow more than one button to be selected         | `boolean`                        | `false`                                  |
| `size`                 | `size`                   | (optional) The size of the button                            | `"large" \| "medium" \| "small"` | `'small'`                                |
| `styles`               | `styles`                 | (optional) Injected CSS styles                               | `string`                         | `undefined`                              |


## Events

| Event          | Description                                                                                        | Type               |
| -------------- | -------------------------------------------------------------------------------------------------- | ------------------ |
| `scale-change` | Emitted when button is clicked                                                                     | `CustomEvent<any>` |
| `scaleChange`  | <span style="color:red">**[DEPRECATED]**</span> in v3 in favor of kebab-case event names<br/><br/> | `CustomEvent<any>` |


## Dependencies

### Depends on

- [scale-helper-text](../helper-text)

### Graph
```mermaid
graph TD;
  scale-segmented-button --> scale-helper-text
  scale-helper-text --> scale-icon-alert-information
  scale-helper-text --> scale-icon-alert-error
  scale-helper-text --> scale-icon-action-success
  style scale-segmented-button fill:#f9f,stroke:#333,stroke-width:4px
```

----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
