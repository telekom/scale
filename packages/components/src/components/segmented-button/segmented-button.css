/**
 * @license
 * Scale https://github.com/telekom/scale
 *
 * Copyright (c) 2021 Egor Kirpichev and contributors, Deutsche Telekom AG
 *
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at https://mozilla.org/MPL/2.0/.
 */

:host {
  --background-color: var(--telekom-color-ui-faint);
  --radius: var(--telekom-radius-standard);
  --height: 32px;

  /* medium styles */
  --height-medium: 40px;

  /* large styles */
  --height-large: 44px;

  display: flex;
  flex-direction: column;
}

.segmented-button {
  background-color: var(--background-color);
  border: 0;
  border-radius: var(--radius);
  padding: 0 var(--telekom-spacing-composition-space-02);
  width: fit-content;
  height: var(--height);
  display: inline-grid;
}

.segmented-button--full-width {
  width: 100%;
}

.segmented-button--medium {
  height: var(--height-medium);
  padding: 0 var(--telekom-spacing-composition-space-02);
}

.segmented-button--large {
  height: var(--height-large);
  padding: 0 var(--telekom-spacing-composition-space-02);
}

.segmented-button--label {
  font-size: var(--telekom-typography-font-size-body);
  font-weight: var(--telekom-typography-font-weight-bold);
  margin-bottom: var(--telekom-spacing-composition-space-04);
}

.segmented-button--helper-text {
  margin-top: var(--telekom-spacing-composition-space-04);
}
