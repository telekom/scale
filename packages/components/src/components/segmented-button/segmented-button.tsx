/**
 * @license
 * Scale https://github.com/telekom/scale
 *
 * Copyright (c) 2021 Egor Kirpichev and contributors, Deutsche Telekom AG
 *
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at https://mozilla.org/MPL/2.0/.
 */

import {
  Component,
  Prop,
  h,
  Host,
  Element,
  State,
  Listen,
  Event,
  EventEmitter,
  Watch,
} from '@stencil/core';
import classNames from 'classnames';
import { emitEvent } from '../../utils/utils';

interface SegmentStatus {
  id: string;
  selected: boolean;
}

@Component({
  tag: 'scale-segmented-button',
  styleUrl: 'segmented-button.css',
  shadow: true,
})
export class SegmentedButton {
  /** segment position within button */
  position = 0;

  slottedSegments = 0;

  @Element() hostElement: HTMLElement;
  /** state */
  @State() status: SegmentStatus[] = [];
  /** (optional) The size of the button */
  @Prop() size?: 'small' | 'medium' | 'large' = 'small';
  /** (optional) Allow more than one button to be selected */
  @Prop() multiSelect: boolean = false;
  /** (optional) If `true`, the button is disabled */
  @Prop({ reflect: true }) disabled?: boolean = false;
  /** (optional) If `true`, expand to container width */
  @Prop() fullWidth?: boolean = false;
  /** (optional) If `true`, show error message */
  @Prop() invalid?: boolean = false;
  /** (optional) If `true`, show error message */
  @Prop() helperText?: string = 'Please select an option';
  /** (optional) Button label */
  @Prop() label?: string;
  /** (optional) icon only */
  @Prop() iconOnly?: boolean = false;  
  /** (optional) Injected CSS styles */
  @Prop() styles?: string;
  /** (optional) aria-label attribute needed for icon-only buttons */
  @Prop()
  ariaLabelTranslation = `segment button with $slottedSegments`;
  @Prop({ mutable: true })
  longestButtonWidth: string;
  /** Emitted when button is clicked */
  @Event({ eventName: 'scale-change' }) scaleChange: EventEmitter;
  /** @deprecated in v3 in favor of kebab-case event names */
  @Event({ eventName: 'scaleChange' }) scaleChangeLegacy: EventEmitter;

  container: HTMLElement;
  showHelperText = false;

  @Listen('scaleClick')
  scaleClickHandler(ev: { detail: { id: string; selected: boolean } }) {
    let tempState: SegmentStatus[];
    if (!this.multiSelect) {
      if (!ev.detail.selected) {
        tempState = this.status.map((obj) =>
          ev.detail.id === obj.id ? ev.detail : { ...obj }
        );
        /* clicked button has now selected state */
      } else {
        tempState = this.status.map((obj) =>
          ev.detail.id === obj.id ? ev.detail : { ...obj, selected: false }
        );
      }
    } else {
      tempState = this.status.map((obj) =>
        ev.detail.id === obj.id ? ev.detail : { ...obj }
      );
    }
    this.setState(tempState);
  }

  @Watch('disabled')
  @Watch('size')
  handlePropsChange() {
    this.propagatePropsToChildren();
  }

  /**
   * Keep props, needed in children buttons, in sync
   */
  propagatePropsToChildren() {
    this.getAllSegmentedButtons().forEach((el) => {
      el.setAttribute('size', this.size);
      if (this.disabled) {
        el.setAttribute('disabled', '');
      }
      if (this.multiSelect) {
        el.setAttribute('multi-select', '');
      }
    });
  }

  connectedCallback() {
    this.propagatePropsToChildren();
  }

  componentDidLoad() {
    const tempState: SegmentStatus[] = [];
    const segmentedButtons = this.getAllSegmentedButtons();
    this.slottedSegments = segmentedButtons.length;
    segmentedButtons.forEach((SegmentedButton) => {
      this.position++;
      tempState.push({
        id: SegmentedButton.getAttribute('segment-id'),
        selected: SegmentedButton.hasAttribute('selected'),
      });
      SegmentedButton.setAttribute('position', this.position.toString());
      SegmentedButton.setAttribute(
        'aria-description-translation',
        '$position $selected'
      );
    });
    this.hostElement.style.setProperty('--colNum', this.slottedSegments.toString())
    this.position = 0;
    this.status = tempState;
    this.setState(tempState);
  }

  componentWillUpdate() {
    this.showHelperText = false;
    if (
      this.invalid &&
      this.status.filter((e) => e.selected === true).length <= 0
    ) {
      this.showHelperText = true;
    }
  }

  getAdjacentSiblings = (tempState, i) => {
    let adjacentSiblings = '';
    if (i !== 0 && tempState[i].selected && tempState[i - 1].selected) {
      adjacentSiblings = 'left';
    }
    if (
      i !== tempState.length - 1 &&
      tempState[i].selected &&
      tempState[i + 1].selected
    ) {
      adjacentSiblings = `${
        adjacentSiblings ? adjacentSiblings + ' right' : 'right'
      }`;
    }
    return adjacentSiblings;
  };

  setState(tempState: SegmentStatus[]) {
    const segmentedButtons = Array.from(
      this.hostElement.querySelectorAll('scale-segment')
    );
    segmentedButtons.forEach((segmentedButton, i) => {
      segmentedButton.setAttribute(
        'adjacent-siblings',
        this.getAdjacentSiblings(tempState, i)
      );
      segmentedButton.setAttribute(
        'selected',
        tempState[i].selected ? 'true' : 'false'
      );
    });
    this.status = tempState;
    emitEvent(this, 'scaleChange', this.status);
  }

  getAllSegmentedButtons() {
    return Array.from(this.hostElement.querySelectorAll('scale-segment'));
  }

  getAriaLabelTranslation() {
    const filledText = this.ariaLabelTranslation.replace(
      /\$slottedSegments/g,
      `${this.slottedSegments}`
    );
    return filledText;
  }

  render() {
    return (
      <Host>
        {this.styles && <style>{this.styles}</style>}
        <fieldset class="segmented-button--fieldset">
          {this.label && (
            <legend class="segmented-button--label"> {this.label} </legend>
          )}
          <ul
            class={this.getCssClassMap()}
            part={this.getBasePartMap()}
            aria-label={this.getAriaLabelTranslation()}
            role="listbox"
            ref={(el) => (this.container = el as HTMLElement)}
            aria-description={
              this.showHelperText && this.helperText ? this.helperText : null
            }            
          >
            <slot />
          </ul>
          {this.showHelperText && (
            <scale-helper-text
              class="segmented-button--helper-text"
              helperText={this.helperText}
              variant={'danger'}
            ></scale-helper-text>
          )}
        </fieldset>

      </Host>
    );
  }

  getBasePartMap() {
    return this.getCssOrBasePartMap('basePart');
  }

  getCssClassMap() {
    return this.getCssOrBasePartMap('css');
  }

  getCssOrBasePartMap(mode: 'basePart' | 'css') {
    const prefix = mode === 'basePart' ? '' : 'segmented-button--';

    return classNames(
      'segmented-button',
      this.size && `${prefix}${this.size}`,
      this.fullWidth && `${prefix}full-width`
    );
  }
}
