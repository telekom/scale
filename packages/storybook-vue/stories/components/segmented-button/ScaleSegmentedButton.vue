<template>
  <scale-segmented-button
    :size="size"
    :disabled="disabled"
    :multi-select="multiSelect"
    :invalid="invalid"
    :label="label"
    :helper-text="helperText"
    :styles="styles"
    @scaleChange="scaleChange"
  >
    <slot></slot>
  </scale-segmented-button>
</template>

<script>
import { action } from '@storybook/addon-actions';
export default {
  props: {
    size: { type: String, default: 'small' },
    disabled: { type: Boolean, default: false },
    multiSelect: { type: Boolean, default: false },
    invalid: { type: Boolean, default: false },
    label: { type: String },
    helperText: { type: String },
    styles: { type: String },
  },
  methods: {
    scaleChange($event) {
      action('scaleChange');
      this.$emit('scaleChange', $event);
    },
    'scale-change'($event) {
      action('scale-change');
      this.$emit('scale-change', $event);
    },
  },
};
</script>
