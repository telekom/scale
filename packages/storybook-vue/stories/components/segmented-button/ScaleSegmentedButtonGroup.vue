<template>
  <scale-segmented-button-group
    :size="size"
    :disabled="disabled"
    :multi-select="multiSelect"
    :styles="styles"
    @scaleChange="scaleChange"
  >
    <slot></slot>
  </scale-segmented-button-group>
</template>

<script>
import { action } from "@storybook/addon-actions";
console.log('FOOOOO')
export default {
  props: {
    size: { type: String, default: 'regular' },
    disabled: { type: Boolean, default: false },
    multiSelect: { type: Boolean, default: false },
    styles: { type: String },
  },
  methods: {
    scaleChange($event) {
      action("scaleChange");
      this.$emit("scaleChange", $event);
    },
    'scale-change'($event) {
      action("scale-change");
      this.$emit("scale-change", $event);
    },
  }
};
</script>