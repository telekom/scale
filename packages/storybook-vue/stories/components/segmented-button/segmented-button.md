<div style="display: inline-flex; align-items: center; justify-content: space-between; width: 100%;">
    <h1>Segmented Button</h1>
    <img src="assets/beta.png" alt="Beta Component" />
</div>

Segmented buttons make it easier for users to filter content, select options, or change views.

## General

A segmented button can consist of two to five selection options, where the options always belong to a common category. Selecting both single and multiple options is possible.

![Image Name](assets/3_components/segmented-button/SegmentedButton-Text_Only_1.png)

### When to use

Use a segmented button to:

- display selected filters,
- offer functions like sorting content or changing views, or
- provide input options.

### When not to use

Don’t use a segmented button when:

- there are more than five selection options,
- the options are not logically related,
- actions should be triggered (e.g., editing, adding, or deleting), or
- the view should switch between separate content (use tabs instead).

## Basic types

![Image Name](assets/3_components/segmented-button/SegmentedButton-Single_Select.png)

### Single selection

As with the Radio Button Group, users can only select one option. One option is always preselected. If necessary, specify a neutral option – for example "All" to override the set filters.

![Image Name](assets/3_components/segmented-button/SegmentedButton-Multiselect.png)

### Multiple selection

As with Check Box Groups, users can also select multiple options. Multiple selections are typically used to filter content.

## Variants

### Text only

![Image Name](assets/3_components/segmented-button/SegmentedButton-Text_Only_1.png)

A text that describes the option is ideal for most use cases. In the selected state, add a checkmark icon to the segmented button.

### Icon and text

![Image Name](assets/3_components/segmented-button/SegmentedButton-Text_Icon.png)

Add appropriate icons to clarify options (optional). In the selected state, a checkmark icon replaces the previously displayed icon.

### Icon only

![Image Name](assets/3_components/segmented-button/Icon_Only.png)

If the icon clearly refers to the action, you can omit the text. In this case, set an aria-label for the screen reader.
By default, icons are used in the outline variant. Flat icons indicate a selected status.

## Sizes

![Image Name](assets/3_components/segmented-button/SegmentedButton-Sizes.png)

Segmented buttons are available in three different sizes:

- Small
- Medium
- Large

The width of the segmented button initially adapts to the length of the text, but can also be switched to full width if necessary. If the width is defined by the text length, all segments fit to the longest segment.

## Disabled state

![Image Name](assets/3_components/segmented-button/SegmentedButton-Disabled.png)

If an interaction is temporarily unavailable, you can deactivate the entire component or just individual segments of the segmented button.

## Elements

![Image Name](assets/3_components/segmented-button/SegmentedButton-Elements-2.png)

### Group label (1) (optional)

Choose a short descriptive title for the whole group (for example, "Sort by" or "Distance").

### Container (2)

The container visually summarizes all the options or fields of a segmented button into a group.

### Segment (3)

Each selection is represented as a segment. With a multiple selection, group adjacent selected segments together.

### Icon (4) (optional)

The icon either supports the label to illustrate the content or replaces the text as a standalone variant.

### Label (5)

The label describes individual choices and should be short and precise.

## Best practice

- Don’t mix variants: text only, text and icon, or icon only.
- Use short and clear labels.
- Offer a neutral option, for example to disable filters and sorting.
- Use the full width option on mobile devices and make sure the total width doesn't exceed the smallest display size (320 px).

## Examples

![Image Name](assets/3_components/segmented-button/SegmentedButton-Filter-list.png)

![Image Name](assets/3_components/segmented-button/SegmentedButton-SortBy-Price.png)

![Image Name](assets/3_components/segmented-button/Map.png)

## Related components

- Tab Navigation
- Button
- Radio Button Group
- Checkbox Group
- Chip
- Icon Button
