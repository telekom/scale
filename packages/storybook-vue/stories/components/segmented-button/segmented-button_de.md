<div style="display: inline-flex; align-items: center; justify-content: space-between; width: 100%;">
    <h1>Segmented Button</h1>
    <img src="assets/beta.png" alt="Beta Component" />
</div>

Segmented Buttons erleichtern Nutzer\*innen das Filtern von Inhalten, die Auswahl von Optionen oder dem Wechsel von Ansichten.

## Allgemein

Ein Segmented Button besteht aus zwei bis fünf Auswahlmöglichkeiten derselben Kategorie. Mit dem Segmented Button kannst du sowohl einzelne Optionen auswählen als auch mehrere gleichzeitig.

![Image Name](assets/3_components/segmented-button/SegmentedButton-Text_Only_1.png)

### Verwende einen Segmented Button, um

- ausgewählte Filter anzuzeigen,
- Funktionen wie Inhalte sortieren oder Ansicht ändern anzubieten oder
- Eingabebeoptionen zur Verfügung zu stellen.

### Verwende Segmented Buttons besser nicht, wenn

- es mehr als fünf Auswahlmöglichkeiten gibt,
- die Optionen in keinem logischen Zusammenhang stehen,
- Aktionen ausgelöst werden sollen (zum Beispiel Editieren, Hinzufügen oder Löschen) oder
- die Ansicht zwischen unterschiedlichen Inhalten gewechselt werden soll (hierfür sind Tabs besser geeignet).

## Basistypen

![Image Name](assets/3_components/segmented-button/SegmentedButton-Single_Select.png)

### Einfachauswahl:

Wie bei der Radio Button Group können Nutzer\*innen nur eine Option auswählen. Es ist immer eine Option vorausgewählt. Gib bei Bedarf eine neutrale Option an – zum Beispiel „Alle“, um gesetzte Filter aufzuheben.

![Image Name](assets/3_components/segmented-button/SegmentedButton-Multiselect.png)

### Mehrfachauswahl:

Wie bei Check Box Groups kannst du auch mehrere Optionen auswählen. Die Mehrfachauswahl wird in der Regel zum Filtern von Inhalten benutzt.

## Varianten

### Nur Text

![Image Name](assets/3_components/segmented-button/SegmentedButton-Text_Only_1.png)

Für die meisten Use Cases reicht ein Text, der die Option beschreibt. Ein Checkmark Icon verdeutlicht den ausgewählten Zustand.

### Icon und Text

![Image Name](assets/3_components/segmented-button/SegmentedButton-Text_Icon.png)

Füge passende Icons hinzu, um die Optionen zu verdeutlichen (optional). Im ausgewählten Zustand ersetzt ein Checkmark Icon das vorher dargestellte Icon.

### Nur Icon

![Image Name](assets/3_components/segmented-button/Icon_Only.png)

Verweist das Icon eindeutig auf die Aktion, kannst du den Text weglassen. Setze in diesem Fall ein Aria-Label für den Screenreader. Icons werden per Default in der Outline-Variante verwendet. Flächige Icons bedeuten, dass das Segment ausgewählt ist.

## Größen

![Image Name](assets/3_components/segmented-button/SegmentedButton-Sizes.png)

Der Segmented Button ist in drei Größen verfügbar:

- Small
- Medium
- Large

Die Breite des Segmented Buttons passt sich der Textlänge an. Bei Bedarf kannst du ihn auch auf volle Breite umschalten. Wenn die Breite durch die Textlänge definiert wird, passen sich alle Segmente dem längsten Segment an.

## Deaktivierter Zustand

![Image Name](assets/3_components/segmented-button/SegmentedButton-Disabled.png)

Du kannst sowohl die gesamte Komponente als auch nur einzelne Segmente des Segmented Buttons deaktivieren, wenn eine Interaktion temporär nicht möglich ist.

## Elemente

![Image Name](assets/3_components/segmented-button/SegmentedButton-Elements-2.png)

### Gruppenlabel (1) (optional)

Wähle einen kurzen unterstützenden Titel für die gesamte Gruppe (zum Beispiel „Sortieren nach“ oder „Distanz“).

### Container (2)

Der Container fasst alle Optionen bzw. die Segmente des Segmented Buttons visuell zu einer Gruppe zusammen.

### Segment (3)

Jede Auswahlmöglichkeit wird als ein Segment dargestellt. Bei der Mehrfachauswahl sind nebeneinanderliegende, ausgewählte Segmente zu einer Gruppe zusammengefasst.

### Icon (4) (optional)

Das Icon unterstützt entweder das Label, um den Inhalt zu veranschaulichen, oder ersetzt den Text als eigenständige Variante.

### Label (5)

Das Label beschreibt einzelne Auswahlmöglichkeiten und sollte kurz und präzise sein.

## Best Practices

- Mische keine Varianten: Nur Text, Text und Icon, Nur Icon.
- Verwende kurze und eindeutige Labels.
- Biete eine neutrale Option, um zum Beispiel Filter und Sortierungen zu deaktivieren.
- Nutze die Full-Width-Option auf Mobilgeräten und stelle sicher, dass die Gesamtbereite nicht die kleinste Displaygröße (320px) überschreitet.

### Beispiele

![Image Name](assets/3_components/segmented-button/SegmentedButton-Filter-list-DE.png)

![Image Name](assets/3_components/segmented-button/SegmentedButton-SortBy-Price-DE.png)

![Image Name](assets/3_components/segmented-button/Map-DE.png)

![Image Name](assets/3_components/segmented-button/SegmentedButton-Tip-DE.png)

## Verwandte Komponenten

- Tab Navigation
- Buttons
- Radio Buttons Group
- Checkbox Group
- Chips
- Icon Button
