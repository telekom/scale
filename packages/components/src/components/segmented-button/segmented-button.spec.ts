import { newSpecPage } from '@stencil/core/testing';
import { SegmentedButton } from './segmented-button';

describe('SegmentedButton', () => {
  it('should match standard snapshot', async () => {
    const page = await newSpecPage({
      components: [SegmentedButton],
      html: `
        <scale-segmented-button>
            <scale-segment>Label</scale-segment>
            <scale-segment>Label</scale-segment>
            <scale-segment>Label</scale-segment>
            <scale-segment>Label</scale-segment>
        </scale-segmented-button>`,
    });
    expect(page.root).toMatchSnapshot();
  });
  it('should include right properties', async () => {
    const page = await newSpecPage({
      components: [SegmentedButton],
      html: `
        <scale-segmented-button>
            <scale-segment>Label</scale-segment>
        </scale-segmented-button>`,
    });
    page.root.invalid = true;
    page.root.fullWidth = true;
    page.root.helperText = 'helpertext';
    page.root.disabled = true;
    page.root.multiSelect = true;
    page.root.size = 'medium';
    await page.waitForChanges();
    expect(page.rootInstance.invalid).toBe(true);
    expect(page.rootInstance.fullWidth).toBe(true);
    expect(page.rootInstance.helperText).toBe('helpertext');
    expect(page.rootInstance.disabled).toBe(true);
    expect(page.rootInstance.multiSelect).toBe(true);
    expect(page.rootInstance.size).toBe('medium');
  });
  it('should match selected button snapshot', async () => {
    const page = await newSpecPage({
      components: [SegmentedButton],
      html: `
        <scale-segmented-button>
            <scale-segment selected>Label</scale-segment>
            <scale-segment selected>Label</scale-segment>
        </scale-segmented-button>`,
    });
    expect(page.root).toMatchSnapshot();
  });
});
