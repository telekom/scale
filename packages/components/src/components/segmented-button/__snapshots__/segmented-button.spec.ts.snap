// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SegmentedButton should match selected button snapshot 1`] = `
<scale-segmented-button>
  <mock:shadow-root>
    <div aria-label="segment button with 2" class="segmented-button segmented-button--small" part="segmented-button small" role="group" style="grid-template-columns: repeat(2, 0px);">
      <slot></slot>
    </div>
  </mock:shadow-root>
  <scale-segment adjacent-siblings="right" aria-description-translation="$position $selected" position="1" selected="true" selected-index="-1" size="small">
    Label
  </scale-segment>
  <scale-segment adjacent-siblings="left" aria-description-translation="$position $selected" position="2" selected="true" selected-index="-1" size="small">
    Label
  </scale-segment>
</scale-segmented-button>
`;

exports[`SegmentedButton should match standard snapshot 1`] = `
<scale-segmented-button>
  <mock:shadow-root>
    <div aria-label="segment button with 4" class="segmented-button segmented-button--small" part="segmented-button small" role="group" style="grid-template-columns: repeat(4, 14px);">
      <slot></slot>
    </div>
  </mock:shadow-root>
  <scale-segment adjacent-siblings="" aria-description-translation="$position $selected" position="1" selected="false" selected-index="-1" size="small">
    Label
  </scale-segment>
  <scale-segment adjacent-siblings="" aria-description-translation="$position $selected" position="2" selected="false" selected-index="-1" size="small">
    Label
  </scale-segment>
  <scale-segment adjacent-siblings="" aria-description-translation="$position $selected" position="3" selected="false" selected-index="-1" size="small">
    Label
  </scale-segment>
  <scale-segment adjacent-siblings="" aria-description-translation="$position $selected" position="4" selected="false" selected-index="-1" size="small">
    Label
  </scale-segment>
</scale-segmented-button>
`;
